README.md

```diff
@@ -396,9 +364,106 @@ Finally, add the `App__peers-count` CSS class to the bottom of `App.css`:
+    } else {
+        publishStateChangeDebounced(todos);
+    }
+});
+```
+
+..and implement the `publishStateChangeDebounced` just after the regular `publishStateChange` like so:
+
+```js
+// src/todos-store.js
+// ...
+import debounce from 'lodash/debounce';
+
+// ...
+const publishStateChangeDebounced = debounce(publishStateChange, 200);
+```
+
+Note that debouncing might cause the UI to not be in sync with the state. That's not an issue in this case because we are using IDs to perform the operations within the store.
+
 ### 8. Deploying the application on IPFS
 
-TODO:
+Instead of using a regular web server to serve the application, we will use IPFS instead. After completing this step, our app will be 100% decentralized!
+
+#### 8.1. Install IPFS and run a local node
+
+We need to have a local IPFS node for the upcoming steps. We will use a [JS IPFS node](https://github.com/ipfs/js-ipfs) but you could use a [Go node](https://github.com/ipfs/go-ipfs) instead.
+
+Let's install it globally:
+
+```sh
+$ npm install -g ipfs
+```
+
+Now, open a new terminal window and start the node:
+
+```sh
+$ jsipfs init
+$ jsipfs daemon
+```
+
+#### 8.2. Ensure links are relative
+
+Everything stored on IPFS is immutable and content-addressable. When a file is stored, IPFS calculates its hash and uses it as an identifier, called `cid`. These files can be accessed in your browser via IPFS node gateways. Because we are running the JS IPFS node, we can access files via `http://localhost:9090/ipfs/<cid>`.
+
+Because the `cid` is unknown at build time, we can't use absolute paths to reference any links or assets. Luckily for us, Create React App allows us to set a `homepage` property which will be used to prefix every asset:
+
+```json
+{
+    "name": "workshop-todo-dapp",
+    "homepage": ".",
+    "...other": "properties"
+}
+```
+
+#### 8.3. Build and deploy
+
+Let's create a production-ready version of the website by building it:
+
+```sh
+$ npm run build
+```
+
+This creates a `build` folder with all the application assets. Let's deploy it to IPFS by adding that folder to your local IPFS node:
+
+```sh
+$ jsipfs add -r build
+```
+
+The `-r` tell `ipfs` to recursively add all the files. At the end of the command output, you should see the `cid` of all added files, including the build folder one:
+
+```
+...
+added QmcFc6EPhavNSfdjG8byaxxV6KtHZvnDwYXLHvyJQPp3uN public/favicon.ico
+added <cid> build
+```
+
+Finally, copy the `<cid>` and use your local IPFS node to access the website: `http://localhost:9090/ipfs/<cid>`. The trick here is that other IPFS nodes that pin the same `<cid>` will also be eligible to serve the website!
+
+#### 8.4. Using a domain
+
+In order to use a domain with your website deployed on IPFS, we need to first understand dnslink. Please watch ["Quick explanation of dnslink in IPFS"](https://www.youtube.com/watch?v=YxKZFeDvcBs) by [@VictorBjelkholm](https://github.com/VictorBjelkholm) that explains what dnslink is in less than 3 minutes.
```

Comment:
Might be a good idea to provide a IPFS hash that some laptop/server within the dweb summit network has pinned. `QmWUcSmgbKV6B8QmiKdfmHG4N438GLkWPjtJwqkJitB6RK` would be the hash.

Reply:
@VictorBjelkholm we did just that in the dweb presentation. After I deployed, @pgte pinned it using the slack bot and I tweeted the URL of the public gateway. Some people actually opened it and added some todos :D